### PR TITLE
DIFM: Fetch themes from the theme showcase

### PR DESCRIPTION
**Pull request by aneeshd16**
#### Changes proposed in this Pull Request

Context: pdh1Xd-GP-p2#comment-953

* We want to let go of our [static theme list](https://github.com/Automattic/wp-calypso/blob/25edea7bc31526d2bcd46d73e56cefb220a3098a/client/signup/steps/difm-design-picker/themes.tsx) and fetch themes from the server. This PR modifies the theme filter for the design picker step to `do-it-for-me` for the DIFM flow.

The `DO NOT MERGE` label has been added to this PR since this needs to be merged only after the `signup/redesigned-difm-flow` feature flag is switched to `true` and all themes listed in pdh1Xd-GP-p2 are tagged with the `do-it-for-me` feature in the theme showcase.

#### Testing instructions

* Go to `/start/do-it-for-me`.
* Select "New Site".
* On the Design Picker step, confirm that themes tagged with the `do-it-for-me` feature are shown. Currently, only one theme (Balasana) has this feature.

Related to 0-as-1201933307121191/f

### CODE REVIEW

File: client/signup/steps/design-picker/index.jsx [hunk 1]
```diff
@@ -74,20 +73,23 @@ export default function DesignPickerStep( props ) {
...
 		data: blockEditorSettings,
-	} = useBlockEditorSettingsQuery( siteId, userLoggedIn && ! props.useDIFMThemes );
+	} = useBlockEditorSettingsQuery( siteId, userLoggedIn && ! props.hideFSEThemes );
 	const isFSEEligible = blockEditorSettings?.is_fse_eligible ?? false;
-	const themeFilters = isFSEEligible
-		? 'auto-loading-homepage,full-site-editing'
-		: 'auto-loading-homepage';
+
+	const getThemeFilters = () => {
+		if ( props.themeFilters ) return props.themeFilters;
```
Comment (p-jackson) [inline client/signup/steps/design-picker/index.jsx]: Another option is to keep the `useDIFMThemes` prop and have `getThemeFilters()` return the hardcoded `'do-it-for-me` filter when it's true. That way we don't need to add an extra prop for excluding FSE themes.

**Review by p-jackson — COMMENTED** [review]
Looks like it's working as expected to me. Balasana is the only theme in the showcase with the `do-it-for-me` flag, so it's the only one that appears. It also has five values in the `theme_subject` taxonomy, which is why there are 5 categories appearing. If you want to tweak those that can be done in the theme showcase too.

Screenshot 2022-03-31 at 12 38 43 PM

Very cool to see the DIFM themes starting to come from the API too 👍

File: client/signup/steps/design-picker/index.jsx [hunk 2]
```diff
@@ -74,20 +73,23 @@ export default function DesignPickerStep( props ) {
...
+
+		return 'auto-loading-homepage';
+	};
 
 	const { data: apiThemes = [] } = useThemeDesignsQuery(
-		{ filter: themeFilters, tier },
+		{ filter: getThemeFilters(), tier },
 		// Wait until block editor settings have loaded to load themes
-		{ enabled: ! props.useDIFMThemes && ! blockEditorSettingsAreLoading }
+		{ enabled: props.hideFSEThemes || ! blockEditorSettingsAreLoading }
```
Comment (arthur791004) [inline client/signup/steps/design-picker/index.jsx]: I think we don't need `props.hideFSEThemes`. We could always wait for block editor settings to be ready and then fetch the themes.

Comment (p-jackson) [reply client/signup/steps/design-picker/index.jsx]: Doesn't it prevent an unnecessary network request in the DIFM flow though?

Comment (arthur791004) [reply client/signup/steps/design-picker/index.jsx]: No, DIFM hides FSE themes and always make network request after this change. I think this flag is try to request theme immediately and we don't need to wait for block editor settings to be ready

Comment (p-jackson) [reply client/signup/steps/design-picker/index.jsx]: Ooooh yes, I misunderstood. You're right we don't need `props.hideFSEThemes` here.

```suggestion
		{ enabled: ! blockEditorSettingsAreLoading }
```

**Review by p-jackson — APPROVED** [review]
LGTM

As expected the design picker makes a network request for the `do-it-for-me` themes and still doesn't make a request for the block editor settings.

Also removes 4KB from the parsed size of the JS bundle 😎

**Review by arthur791004 — APPROVED** [review]

**Comment by aneeshd16** [comment]
Rebased after #62400 was merged.

**Pull request merged**